src/Catalog/CatalogIndexEntry.cs

```diff
@@ -28,65 +25,49 @@ public CatalogIndexEntry(JToken item)
                 NuGetVersion.Parse(item["nuget:version"].ToString()))
         {
         }
-
+        
         public CatalogIndexEntry(Uri uri, string type, string commitId, DateTime commitTs, string id, NuGetVersion version)
         {
-            _uri = uri;
+            Uri = uri;
             _type = type;
-            _commitId = commitId;
-            _commitTimeStamp = commitTs;
-            _id = id;
-            _version = version;
+            CommitId = commitId;
+            CommitTimeStamp = commitTs;
+            Id = id;
+            Version = version;
         }
 
-        public Uri Uri
+        [JsonConstructor]
```

Comment:
No, as long as the set method exists (regardless of if it is private or not), `JSON.NET` will use it.